app/javascript/react/components/NewMovieForm.js

```diff
@@ -0,0 +1,59 @@
+import React, {useState} from "react"
+import NewMovieFormTile from "./NewMovieFormTile"
+import { Redirect } from "react-router-dom"
+
+const NewMovieForm = (props) => {
+
+  const [success, setSuccess] = useState({movieSaved: false, response: []})
+
+  const addMovie = async (formPayload) => {
+    try {
+      const response = await fetch("/api/v1/movies", {
+        credentials: "same-origin",
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "Accept": "application/json"
+        },
+        body: JSON.stringify(formPayload)
+      })
+      if(!response.ok){
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error)
+      }
+      const responseBody = await response.json()
+      setSuccess(responseBody)
+    } catch(error) {
+      console.log(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  const redirectSwitch = () => {
```

Comment:
this doesn't need to be a function (a function means we have to invoke it within the code) where we can instead define the "free floating" `if` logic here in this component